feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/start/setupStakingType/SetupStakingTypeSelectionMixin.kt

```diff
@@ -0,0 +1,69 @@
+package io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupStakingType
+
+import io.novafoundation.nova.feature_staking_api.domain.model.Validator
+import io.novafoundation.nova.feature_staking_impl.domain.nominationPools.model.NominationPool
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.RecommendableMultiStakingSelection
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.SelectionTypeSource
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.StartMultiStakingSelection
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.store.StartMultiStakingSelectionStoreProvider
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.store.currentSelectionFlow
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupAmount.direct.DirectStakingSelection
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupAmount.pools.NominationPoolSelection
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.first
+
+class SetupStakingTypeSelectionMixinFactory(
+    private val currentSelectionStoreProvider: StartMultiStakingSelectionStoreProvider,
+    private val editableSelectionStoreProvider: StartMultiStakingSelectionStoreProvider,
+) {
+
+    fun create(
+        scope: CoroutineScope
+    ): SetupStakingTypeSelectionMixin {
+        return SetupStakingTypeSelectionMixin(
+            currentSelectionStoreProvider,
+            editableSelectionStoreProvider,
+            scope
+        )
+    }
+}
+
+class SetupStakingTypeSelectionMixin(
+    private val currentSelectionStoreProvider: StartMultiStakingSelectionStoreProvider,
+    private val editableSelectionStoreProvider: StartMultiStakingSelectionStoreProvider,
+    private val scope: CoroutineScope
+) {
+
+    val editableSelectionFlow = editableSelectionStoreProvider.currentSelectionFlow(scope)
+        .filterNotNull()
+
+    // TODO use it in SetupStakingTypeViewModel after refactoring
```

Comment:
Pending TODO

Reply:
Remember for that. To avoid conflicts I will close it after merging branches